trio/_core/_run.py

```diff
@@ -530,6 +530,7 @@ def _close(self, exc):
 
     @enable_ki_protection
     def __exit__(self, etype, exc, tb):
+        # locals()["@TRIO_KI_PROTECTION_ENABLED"]=True
```

Comment:
need to resolve this (remove, uncomment, or document)

Reply:
Oops! 😬 I'll tidy that up when I push the inlined ki protection.